src/components/AttachmentModal.tsx

```diff
@@ -1,189 +1,211 @@
-    onModalHide: PropTypes.func,
+type Attachment = {
+    source: string;
+    isAuthTokenRequired: boolean;
+    file: File;
+    isReceipt: boolean;
+};
 
-    /** Optional callback to fire when we want to do something after attachment carousel changes. */
-    onCarouselAttachmentChange: PropTypes.func,
+type ImagePickerResponse = {
+    height: number;
+    name: string;
+    size: number;
+    type: string;
+    uri: string;
+    width: number;
+};
 
-    /** Optional original filename when uploading */
-    originalFileName: PropTypes.string,
+type FileObject = File | ImagePickerResponse;
 
-    /** A function as a child to pass modal launching methods to */
-    children: PropTypes.func,
+type ChildrenProps = {
+    displayFileInModal: (data: FileObject) => void;
+    show: () => void;
+};
+
+type AttachmentModalProps = AttachmentModalOnyxProps & {
+    /** Optional source (URL, SVG function) for the image shown. If not passed in via props must be specified when modal is opened. */
+    source?: AvatarSource;
+
+    /** Optional callback to fire when we want to preview an image and approve it for use. */
+    onConfirm?: ((file: Partial<File>) => void) | null;
+
+    /** Whether the modal should be open by default */
+    defaultOpen?: boolean;
+
+    /** Optional original filename when uploading */
+    originalFileName?: string;
 
     /** Whether source url requires authentication */
-    isAuthTokenRequired: PropTypes.bool,
+    isAuthTokenRequired?: boolean;
 
     /** Determines if download Button should be shown or not */
-    allowDownload: PropTypes.bool,
+    allowDownload?: boolean;
 
     /** Title shown in the header of the modal */
-    headerTitle: PropTypes.string,
+    headerTitle?: string;
 
     /** The report that has this attachment */
-    report: reportPropTypes,
+    report?: OnyxTypes.Report;
 
-    /** The transaction associated with the receipt attachment, if any */
-    transaction: transactionPropTypes,
+    /** Optional callback to fire when we want to do something after modal show. */
+    onModalShow?: () => void;
 
-    ...withLocalizePropTypes,
+    /** Optional callback to fire when we want to do something after modal hide. */
+    onModalHide?: () => void;
 
-    ...windowDimensionsPropTypes,
+    /** Optional callback to fire when we want to do something after attachment carousel changes. */
+    onCarouselAttachmentChange?: (attachment: Attachment) => void;
 
     /** Denotes whether it is a workspace avatar or not */
-    isWorkspaceAvatar: PropTypes.bool,
+    isWorkspaceAvatar?: boolean;
 
     /** Whether it is a receipt attachment or not */
-    isReceiptAttachment: PropTypes.bool,
+    isReceiptAttachment?: boolean;
 
-    /** Whether the receipt can be replaced */
-    canEditReceipt: PropTypes.bool,
-};
+    /** A function as a child to pass modal launching methods to */
+    children?: React.FC<ChildrenProps>;
```

Comment:
Could you test if this prop works when using the component?

Reply:
@fabioh8010 what do you mean by that? 
If that the prop have some values when using component ? 
or just if this component work when we pass children to it? 
overall both cases yes :D 
<img width="1121" alt="image" src="https://github.com/Expensify/App/assets/25230417/52677dd2-2890-47aa-b95e-6433f6c2fe46">

Reply:
I meant the TS part, e.g. Use this component in a new file and pass the children props, or convert a component that use this one to TS and see if this prop is properly typed.

Reply:
Yep, its working 
<img width="965" alt="image" src="https://github.com/Expensify/App/assets/25230417/35d2093d-b170-464a-aabe-40cac3fc36be">